core/iwasm/interpreter/wasm_interp_fast.c

```diff
@@ -1128,12 +1128,27 @@ wasm_interp_dump_op_count()
         goto *p_label_addr;                            \
     } while (0)
 #else
-#define FETCH_OPCODE_AND_DISPATCH()                                 \
-    do {                                                            \
-        const void *p_label_addr = label_base + *(int16 *)frame_ip; \
-        frame_ip += sizeof(int16);                                  \
-        goto *p_label_addr;                                         \
+#if UINTPTR_MAX == UINT64_MAX
+#define FETCH_OPCODE_AND_DISPATCH()                                       \
+    do {                                                                  \
+        const void *p_label_addr;                                         \
+        bh_assert(((uintptr_t)frame_ip & 1) == 0);                        \
```

Comment:
should `& 3`?

Reply:
No, the frame_ip must be 2-byte aligned, but not necessarily 4-byte aligned. If we force the frame_ip to be 4-byte aligned in wasm_loader, we need to add padding byte if needed, and here, we need to add code like `frame_ip = align_uint(frame_ip, 4)` or `if (frame_ip & 2) frame_ip += 2`, I had a try, it impacts performance a lot.

Reply:
Got it, Thanks!

I originally thought the CPU requires 4-bytes align, if 2-bytes align is OK, then no problem here.